include/functions.php

```diff
@@ -91,3 +91,20 @@ function pll_use_block_editor_plugin() {
 	 */
 	return class_exists( 'PLL_Block_Editor_Plugin' ) && apply_filters( 'pll_use_block_editor_plugin', ! defined( 'PLL_USE_BLOCK_EDITOR_PLUGIN' ) || PLL_USE_BLOCK_EDITOR_PLUGIN );
 }
+
+/**
+ * Strips out all characters not allowed in a locale code.
+ *
+ * @since 3.5
+ *
+ * @param string $locale_code The locale code to be sanitized.
+ * @return string The sanitized value.
+ */
+function pll_sanitize_locale_code( $locale_code ) {
```

Comment:
I would prefer a polyfill. This is then easier to maintain when we stop supporting old versions. I just have to remove the polyfill without caring wher the function's used as I did in https://github.com/polylang/polylang/commit/925366af3746e575660816911578339136652aab

Reply:
Makes sense.
Done in [08238b6](https://github.com/polylang/polylang/pull/1280/commits/08238b644d453f4bb4e45d9635eb5df5a1803a5a).

However, I didn't strictly copied the original function, I added a type-cast `(string) preg_replace( ... )` because PHPStan wasn't happy: `preg_replace()` can return `null` on error.
I chose this instead of adding an exception in `phpstan.neon.dist` because:
- There shouldn't have such errors.
- Easier to remove.